crates/bindings/src/lib.rs

```diff
@@ -65,23 +66,58 @@ pub struct ReducerContext {
     /// For automatic reducers, i.e. `init`, `update` and scheduled reducers,
     /// this will be the module's `Address`.
     pub address: Option<Address>,
+    pub db: Local,
 }
 
 impl ReducerContext {
     #[doc(hidden)]
     pub fn __dummy() -> Self {
         Self {
+            db: Local {},
             sender: Identity::__dummy(),
             timestamp: Timestamp::UNIX_EPOCH,
             address: None,
         }
     }
 }
 
+/// A handle on a database with a particular table schema.
+pub trait DbContext {
+    /// A view into the tables of a database.
+    ///
+    /// This type is specialized on the database's particular schema.
+    ///
+    /// Methods on the `DbView` type will allow querying tables defined by the module.
+    type DbView;
+
+    /// Get a view into the tables.
+    ///
+    /// This method is provided for times when a programmer wants to be generic over the `DbContext` type.
+    /// Concrete-typed code is expected to read the `.db` field off the particular `DbContext` implementor.
+    /// Currently, being this generic is only meaningful in clients,
+    /// as modules have only a single implementor of `DbContext`.
+    fn db(&self) -> &Self::DbView;
+}
```

Comment:
`DbContext` in modules also has methods `rng`, `identity` and `address`. `identity` and `address` are subject to change and need not be implemented yet, but `rng` is important and should be included.

Reply:
If we add the trait later and move inherent methods to trait methods, that's a breaking change.

Reply:
Well, it wouldn't necessarily be moving them - we could just keep the inherent methods and add the trait impl. From a user perspective, having to always import this weird `DbContext` trait just to use an rng seems unergonomic. I think using trait syntax to define this API is really helpful, but that doesn't mean we have to have a singleton trait with only one impl in the actual crate.

Reply:
Does the same reasoning apply to clients, where you have to import the `DbContext` trait in order to do things like subscribe to queries and inspect the connection's status?